chapter4.adoc

```diff
@@ -123,107 +145,71 @@ domain
+the original access type.
 
-* 00b - `1G_disallow` - access to the 1G range is disallowed
-* 01b - `1G_allow` - access to the 1G range is allowed
-* 10b - `MTT_L1_DIR` - The 64M range is composed of 16384x4K pages
-* 11b - `2M_PAGES` - The 64M range is composed of 32x2M pages
+4. Otherwise, the _mpte_ is valid. If (_i_=0) or (_i_=1 and _mpte.type_ is not
+`MTT_L1_DIR`), go to step 5. Otherwise, the _mpte_ is a pointer to the next
+level of the `MTT`. Let _i_ = _i_-1. If _i_ < 0, stop and raise an access-fault
+exception corresponding to the original access type. Otherwise, let
+_a_ = _mpte.ppn_ x PAGESIZE and go to step 2. Note that when _mpte.type_ =
+`MTT_L1_DIR`, the _mpte.ppn_ field is the value of the _mpte.info_ field.
 
-The `INFO` field depends on the `TYPE` field and is formatted as per the table:
+5. A leaf _mpte_ has been found. If any bits or encodings within _mpte.type_
+and _mpte.info_ that are reserved for future standard use, per
+<<Smmtt-rw-l2-encoding>>, are set within _mpte_, stop and raise an access-fault
+exception corresponding to the access type.
 
-.`MTTL2` Entry for Smmtt[34, 46, 56]
-[width="100%",cols="25%,75%",options="header",]
-|===
-|*`MTTL2` Entry Type* |*Description, `INFO` and `TYPE` field encoding*
-|`1G_allow` a|
-_The 1G range of address is allowed for the domain._
-The `INFO` field must be 0.
-When configuring 1G ranges, RDSM ensures that 16 `MTTL2` entries, each
-corresponding to 64M of address space, have identical `TYPE` field values.
+6. The _mpte_ is a valid leaf _mpte_. Fetch the access-permissions for the
+physical address per the steps described below:
 
-|`1G_disallow` a|
-_The 1G range of address is not allowed for the domain._
-The `INFO` field must be 0.
-When configuring 1G ranges, RDSM ensures that 16 `MTTL2` entries, each
-corresponding to 64M of address space, have identical `TYPE` field values.
+* if _i_=1, and the _mpte.type_ field directly specifies the access-permissions
+for 1GB page regions (via 32 MTTL2 entries with identical _mpte.type_ values -
+see <<Smmtt-rw-l2-encoding>>); go to step 7, else
 
-|`MTT_L1_DIR` a|
-The `INFO` field provides the PPN of the `MTTL1` page.
-Entries of the `MTTL1` page hold a 2-bit `PERM` field to indicate the access
-for the supervisor domain (described in the `MTTL1` entry <<mtt-l1>>).
+* if _i_=1, and the _mpte.type_ field value is `2M_PAGES`, then the
+_mpte.info_[31:0] field contains 16 entries of 2-bit access-permissions encoding
+for 16 2M address regions (see <<Smmtt-rw-l2-encoding>>); go to step 7, else
 
-|`2M_PAGES` a|
-_The 64M range of address space is partitioned into 2M pages where each
-page has access allowed/not._
-The `INFO` field bits 31:0 holds 1 bit per 2M address range to indicate
-access disallowed(0b) or allowed (1b). `INFO` field bits 43:32 are
-reserved (must be zero).
-|===
+* If _i_=0, the _mpte_ contains a 4-bit entry that holds the access-permission
```

Comment:
The mpte actually contains 16x4-bit entries. May want to describe this as a shift and mask or say extract bits i:j where i and j are determined by pa[0].